pkg/apis/cosigned/v1alpha1/clusterimagepolicy_validation.go

```diff
@@ -144,6 +147,39 @@ func (source *Source) Validate(ctx context.Context) *apis.FieldError {
 	return errs
 }
 
+func (a *Attestation) Validate(ctx context.Context) *apis.FieldError {
+	var errs *apis.FieldError
+	if a.Name == "" {
+		errs = errs.Also(apis.ErrMissingField("name"))
+	}
+	if a.PredicateType == "" {
+		errs = errs.Also(apis.ErrMissingField("predicateType"))
+	} else if a.PredicateType != "custom" && a.PredicateType != "slsaprovenance" && a.PredicateType != "spdx" && a.PredicateType != "link" && a.PredicateType != "vuln" {
```

Comment:
seems like a good use of `switch`?

Reply:
Doesn't `cosign attest` support taking a full predicate type URL as well?  Should we also allow full-blown URLs?

Reply:
I'd rather not pollute this with the switch only because I'd rather use the TODO (refactor where the options come from and use that), also attest may take the uri but verify-attest does not. It looks for the short name in the map and pulls the URL from it. Again, I think that should be done as part of a follow on PR to refactor that code a bit to pull the map out of the cli/options so it can be reused here.
https://github.com/sigstore/cosign/blob/main/pkg/policy/attestation.go#L43